### PR TITLE
Cmr 479

### PR DESCRIPTION
**Pull request by dseelig**

### CODE REVIEW

File: src/main/java/de/qucosa/routes/Main.java [hunk 1]
```diff
@@ -27,22 +34,55 @@
...
-                .to("mock:test");
+                .aggregate(constant(true), recordListAggragator).completionSize(2)
+                .to("oaiprovider:update")
+                .log("${body}");
+
+        from("direct:dcdiss")
+                .id("build-dc-dissemination")
+                .setProperty("transfer.url.pattern", simple("{{transfer.url.pattern}}"))
+                .setProperty("xsltStylesheetResourceName", simple("/xslt/mets2dcdata.xsl"))
+                .setProperty("agentNameSubstitutions", simple(""))
```
Comment (claussni) [inline src/main/java/de/qucosa/routes/Main.java]: Here, and below, another configuration parameter is needed , e.g. ```agent.name.substitutions```

File: src/main/java/de/qucosa/routes/Main.java [hunk 2]
```diff
@@ -54,4 +94,13 @@ public void configure() {
 
     }
 
+    private Namespaces namespaces() {
+        Namespaces namespaces = new Namespaces("", "");
```
Comment (claussni) [inline src/main/java/de/qucosa/routes/Main.java]: I'm unsure about this constructor. Empty prefix and URL make no sense. Is there another one?

File: src/main/java/de/qucosa/routes/Main.java [hunk 3]
```diff
@@ -54,4 +94,13 @@ public void configure() {
 
     }
 
+    private Namespaces namespaces() {
```
Comment (claussni) [inline src/main/java/de/qucosa/routes/Main.java]: This method returns the same Map every time. This surly can be moved out to a local variable.

File: src/main/java/de/qucosa/transformers/AbstractDisseminationTransform.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+abstract public class AbstractDisseminationTransform {
+    protected DissTerms dissTerms = new DissTerms();
```
Comment (claussni) [inline src/main/java/de/qucosa/transformers/AbstractDisseminationTransform.java]: This member variables are never used and should be removed.

File: src/main/java/de/qucosa/transformers/AbstractDisseminationTransform.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import org.w3c.dom.Document;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+abstract public class AbstractDisseminationTransform {
```
Comment (claussni) [inline src/main/java/de/qucosa/transformers/AbstractDisseminationTransform.java]: This class has no abstract methods. Why should it be abstract?
If it's a base class helping with METS XML extraction, I suggest you name it ```MetsSupport``` and don't use it as a base class.

File: src/main/java/de/qucosa/routes/Main.java (hunk 2 above)
Comment (dseelig) [reply src/main/java/de/qucosa/routes/Main.java]: Is not given an empty standard constructor in namespaces class.

**Pull request merged**